src/profile-logic/import/chrome.js

```diff
@@ -0,0 +1,363 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import type {
+  Profile,
+  StackTable,
+  Thread,
+  IndexIntoFuncTable,
+  IndexIntoStackTable,
+} from '../../types/profile';
+
+import { getEmptyProfile } from '../../profile-logic/profile-data';
+import { getEmptyThread } from '../../test/fixtures/profiles/make-profile';
+import { ensureExists } from '../../utils/flow';
+
+// Chrome Tracing Event Spec:
+// https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+type TracingEventUnion = ProfileChunkEvent | ThreadNameEvent;
+
+type TracingEvent<Event> = {|
+  cat: string,
+  id: string,
+  ph: string, // Phase
+  pid: number, // Process ID
+  tid: number, // Thread ID
+  ts: number, // Timestamp
+  ...Event,
+|};
+
+type ProfileChunkEvent = TracingEvent<{|
+  name: 'ProfileChunk',
+  args: {
+    data: {
+      cpuProfile: {
+        nodes?: Array<{
+          callFrame: {
+            functionName: string,
+            scriptId: number,
+            lineNumber?: number,
+            columnNumber?: number,
+            url?: string,
+          },
+          id: number,
+          parent?: number,
+        }>,
+        samples: number[], // Index into cpuProfile nodes
+      },
+      timeDeltas: number[],
+    },
+  },
+  phase: 'P',
+|}>;
+
+type ThreadNameEvent = TracingEvent<{|
+  name: 'thread_name',
+  ph: 'm',
+  args: { name: string },
+|}>;
+
+type ScreenshotEvent = TracingEvent<{|
+  name: 'Screenshot',
+  ph: 'O',
+  args: { snapshot: string },
+|}>;
+
+export function isChromeProfile(profile: mixed): boolean {
+  if (!Array.isArray(profile)) {
+    return false;
+  }
+  const event = profile[0];
+  // Lightly check that some properties exist that are in the TracingEvent.
```

Comment:
It's weird that they don't have any metadata in their profile 😕

Reply:
Yeah I just double checked. I didn't see anything better in there to key off of.